shell/browser/api/electron_api_top_level_window.h

```diff
@@ -259,6 +259,9 @@ class TopLevelWindow : public gin_helper::TrackableObject<TopLevelWindow>,
 
   std::unique_ptr<NativeWindow> window_;
 
+  // Reference to JS wrapper to prevent garbage collection.
+  v8::Global<v8::Value> self_ref_;
```

Comment:
It seems like this never gets set to anything... given the tests pass, that means one of:

1. The tests aren't actually testing that the window is retained, or
2. There's a different cycle that this PR hasn't broken.